accounts/views.py

```diff
@@ -8,36 +8,34 @@
 from django.shortcuts import redirect
 from django.utils.decorators import method_decorator
 from django.views.decorators.csrf import csrf_exempt
+from django.views.generic.base import View
 from oauth2_provider.models import get_access_token_model
 from oauth2_provider.views import IntrospectTokenView
-from rest_framework import generics
-from rest_framework_api_key.permissions import HasAPIKey
 from sentry_sdk import capture_message
 
 from accounts.auth import LabsView, PennView
 from accounts.serializers import UserSerializer
 
 
-class LoginView(generics.GenericAPIView):
+class LoginView(View):
     """
     Log in a user.
+    WARNING: You must ensure this page is protected by Shibboleth and Clean Headers
+    See https://github.com/nginx-shib/nginx-http-shibboleth
     """
-    def get(self, request):
-        # Validate API Key
-        if not HasAPIKey.has_permission(self, request, self):
-            return redirect('https://auth.pennlabs.org/login/?next=' + request.GET.get('next', ''))
 
-        # API is valid, login user
+    def get(self, request):
         pennkey = request.META.get('HTTP_EPPN', '').lower().split('@')[0]
         first_name = request.META.get('HTTP_GIVENNAME', '').lower().capitalize()
         last_name = request.META.get('HTTP_SN', '').lower().capitalize()
         email = request.META.get('HTTP_MAIL', '').lower()
-        shibboleth_attributes = {'first_name': first_name, 'last_name': last_name, 'email': email}
+        affiliation = request.META.get('HTTP_UNSCOPED_AFFILIATION', '').split(';')
+        shibboleth_attributes = {'first_name': first_name, 'last_name': last_name, 'email': email,
+                                 'affiliation': affiliation}
         user = auth.authenticate(remote_user=pennkey, shibboleth_attributes=shibboleth_attributes)
         if user:
             auth.login(request, user)
-            params = request.get_full_path().split('next=')[1]
-            return redirect('https://platform.pennlabs.org' + params)
+            return redirect(request.GET.get('next', '/'))
```

Comment:
I would protect this using a whitelist, since it could possibly open us up to phishing attacks by making the URL appear to come from us.

It might be a bit difficult, since there are a lot of different products and domains we could use this with. Maybe a simple list of domain names for now and expanding that into a model later?

Reply:
I don't believe the redirect exposes us to any potential attack vectors. the `user.authenticate` method creates a cookie valid only on `platform.pennlabs.org` to verify the user is logged in, that information doesn't get sent through the redirect.